application/src/Stdlib/Environment.php

```diff
@@ -0,0 +1,86 @@
+<?php
+namespace Omeka\Stdlib;
+
+use Zend\ServiceManager\ServiceLocatorInterface;
+
+class Environment
+{
+    /**
+     * The PHP minimum version
+     */
+    const PHP_MINIMUM_VERSION = '7.1.0';
+
+    /**
+     * The MySQL minimum version
+     */
+    const MYSQL_MINIMUM_VERSION = '5.6.4';
+
+    /**
+     * The required PHP extensions
+     */
+    const PHP_REQUIRED_EXTENSIONS = ['fileinfo', 'mbstring', 'PDO', 'pdo_mysql', 'xml'];
+
+    /**
+     * @var array Environment error messages
+     */
+    protected $errorMessages = [];
+
+    /**
+     * @param ServiceLocatorInterface $services
+     */
+    public function __construct(ServiceLocatorInterface $services)
+    {
+        $translator = $services->get('MvcTranslator');
```

Comment:
Perhaps we should be using the Message container for these and just translating on output in the installer and upgrader

Reply:
Why not just translate them here, at the point where they're set?

Reply:
we mostly translate at print time, largely to avoid doing what this does: injecting the translator "deep" into services

this service could then also be constructed with just the Connection directly rather than taking the service manager, as that's the only service it uses (of course this could be true now but still)

Reply:
We shouldn't pass the Connection into the constructor because the Environment tests if the Connection factory throws an error. If we pass the connection, the Connection will have already been created (or attempted to be created).

Reply:
hmm, will the connection actually be created at that call time ever? I want to say it won't... it's actually `connect()` that we're catching errors in, no?

Reply:
Interesting point. I assumed the installer task fetched the Connection in the `try...catch` because it needed to check if the factory was throwing. If that's not the case (and I suppose it's not) then I'll pass the Connection to the constructor.

Reply:
I think it's a case where old assumptions were that that _would_ be the case (and maybe it even was in earlier versions... though I doubt it somewhat). But now with the Status service and even moreso the ModuleManager (since it gets the Connection right away before it decides whether or not to try using it), the service will have failed or been created well before the installer or upgrader goes to check it.